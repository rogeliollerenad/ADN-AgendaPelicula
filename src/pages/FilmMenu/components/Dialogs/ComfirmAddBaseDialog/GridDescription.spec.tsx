/* eslint-disable react/jsx-props-no-spreading, @typescript-eslint/no-var-requires, global-require */

import { render } from '@testing-library/react';
import MockDate from 'mockdate';
import {
  GridDescription,
  PropsDataDescription,
} from 'pages/FilmMenu/components/Dialogs/ComfirmAddBaseDialog/GridDescription';
import React from 'react';

jest.mock('react-intersection-observer', () => {
  const { factory } = require('tests/mocks/react-intersection-observer.mock');
  return factory();
});

describe('GridDescription Test', () => {
  let props: PropsDataDescription;

  beforeAll(() => {
    MockDate.set('2020-10-31');
  });

  afterAll(() => {
    MockDate.reset();
  });

  beforeEach(() => {
    props = {
      description: 'Lorem Ipsum',
      title: 'Titulo Testing',
      released: '10/20/2000',
      time: 120,
      score: '3.5',
      price: 10,
    };
  });

  it('should match snapshot', () => {
    const { container } = render(<GridDescription {...props} />);
    expect(container).toMatchSnapshot();
  });
});
