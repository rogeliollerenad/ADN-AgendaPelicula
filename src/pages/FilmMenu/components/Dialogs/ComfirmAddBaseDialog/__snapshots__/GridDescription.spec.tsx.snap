// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridDescription Test should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root dialog-card-title MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Titulo Testing
      </h2>
      <p
        class="MuiTypography-root dialog-card-description MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Lorem Ipsum
      </p>
      <div
        class="MuiGrid-root dialog-card-details-wrap MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <p
            class="MuiTypography-root dialog-card-details-items MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 38.013c-22.458 0-66.472 110.3-84.64 123.502-18.17 13.2-136.674 20.975-143.614 42.334-6.94 21.358 84.362 97.303 91.302 118.662 6.94 21.36-22.286 136.465-4.116 149.665 18.17 13.2 118.61-50.164 141.068-50.164 22.458 0 122.9 63.365 141.068 50.164 18.17-13.2-11.056-128.306-4.116-149.665 6.94-21.36 98.242-97.304 91.302-118.663-6.94-21.36-125.444-29.134-143.613-42.335-18.168-13.2-62.182-123.502-84.64-123.502z"
              />
            </svg>
             
            3.5
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
        >
          <p
            class="MuiTypography-root dialog-card-details-items MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M208.242 24.629l-52.058 95.205 95.207 52.059 17.271-31.586-42.424-23.198A143.26 143.26 0 0 1 256 114c78.638 0 142 63.362 142 142s-63.362 142-142 142-142-63.362-142-142c0-16.46 2.785-32.247 7.896-46.928l-32.32-16.16C82.106 212.535 78 233.798 78 256c0 98.093 79.907 178 178 178s178-79.907 178-178S354.093 78 256 78c-13.103 0-25.875 1.44-38.18 4.148l22.008-40.25-31.586-17.27zm104.27 130.379L247 253.275V368h18V258.725l62.488-93.733-14.976-9.984z"
              />
            </svg>
             
            120
             Â  Min.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <p
            class="MuiTypography-root dialog-card-details-items MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M119 31v66h18V31h-18zm256 0v66h18V31h-18zM73 89v30h366V89h-28v26h-54V89H155v26h-54V89H73zm0 48v302h366V137H73zm110 14h18v32h46v-32h18v32h46v-32h18v32h46v-32h18v32h32v18h-32v46h32v18h-32v46h32v18h-32v46h32v18H265v32h-18v-32h-46v32h-18v-32h-46v32h-18v-32H87v-18h32v-46H87v-18h32v-46H87v-18h32v-46H87v-18h96v-32zm-46 50v46h46v-46h-46zm64 0v46h46v-46h-46zm64 0v46h46v-46h-46zm64 0v46h46v-46h-46zm-192 64v46h46v-46h-46zm64 0v46h46v-46h-46zm64 0v13.27c2.545-5.936 7.334-10.725 13.27-13.27H265zm32.73 0c5.936 2.545 10.725 7.334 13.27 13.27V265h-13.27zm31.27 0v46h46v-46h-46zm-41 16c-3.973 0-7 3.027-7 7s3.027 7 7 7 7-3.027 7-7-3.027-7-7-7zm-23 16.73V311h13.27c-5.936-2.545-10.725-7.334-13.27-13.27zm46 0c-2.545 5.936-7.334 10.725-13.27 13.27H311v-13.27zM137 329v46h46v-46h-46zm64 0v46h46v-46h-46zm64 0v46h46v-46h-46zm64 0v46h46v-46h-46z"
              />
            </svg>
            <time
              datetime="972018000000"
            >
              20 de oct. de 2000
            </time>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="justify-content: space-around;"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            data-cy="select-day"
            data-testid="day_picker"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              for="date-picker-inline"
              id="date-picker-inline-label"
            >
              Selecciona Fecha
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                id="date-picker-inline"
                type="text"
                value="10/30/2020"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            data-cy="select-hour"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              for="time-picker"
              id="time-picker-label"
            >
              Selecciona Hora
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                id="time-picker"
                type="text"
                value="07:00 PM"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change time"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="MuiGrid-root dialog-card-wrap-price MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
          >
            <p
              class="MuiTypography-root dialog-card-price MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              <span
                style="color: rgb(0, 0, 0);"
              >
                Precio Final: 
              </span>
              $10.00
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
